internal/schemarepo/loadschemas/plugins.go

```diff
@@ -141,6 +137,30 @@ func (cp *Plugins) ProviderSchema(addr addrs.Provider) (providers.ProviderSchema
 		if r.Version < 0 {
 			return resp, fmt.Errorf("provider %s has invalid negative schema version for managed resource type %q, which is a bug in the provider", addr, t)
 		}
+
+		// Validate resource identity schema if the resource has one
+		if r.Identity != nil {
+			if err := r.Identity.InternalValidate(); err != nil {
+				return resp, fmt.Errorf("provider %s has invalid identity schema for managed resource type %q, which is a bug in the provider: %q", addr, t, err)
+			}
+			if r.IdentityVersion < 0 {
+				return resp, fmt.Errorf("provider %s has invalid negative identity schema version for managed resource type %q, which is a bug in the provider", addr, t)
+			}
+
+			for attrName, attrTy := range r.Identity.ImpliedType().AttributeTypes() {
+				if attrTy.MapElementType() != nil {
```

Comment:
We may be able to consolidate all these type checks with `if attrTy.IsCollectionType()`